### PR TITLE
Add node resources methods to gcs server node info handler

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
This is the seventh pr of gcs server which add node resources methods to gcs server node info handler.

## Related issue number
https://github.com/ray-project/ray/pull/6401

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: src/ray/rpc/gcs_server/gcs_rpc_client.h [hunk 1]
```diff
@@ -84,6 +84,18 @@ class GcsRpcClient {
   VOID_RPC_CLIENT_METHOD(NodeInfoGcsService, ReportBatchHeartbeat, request, callback,
                          node_info_grpc_client_)
 
+  /// Get node's dynamic resources from GCS Service.
```
Comment (raulchen) [inline src/ray/rpc/gcs_server/gcs_rpc_client.h]:
```suggestion
  /// Get node's resources from GCS Service.
```

File: src/ray/rpc/gcs_server/gcs_rpc_client.h [hunk 2]
```diff
@@ -84,6 +84,18 @@ class GcsRpcClient {
...
+  VOID_RPC_CLIENT_METHOD(NodeInfoGcsService, GetResources, request, callback,
+                         node_info_grpc_client_)
+
+  /// Update dynamic resources of node in GCS Service.
```
Comment (raulchen) [inline src/ray/rpc/gcs_server/gcs_rpc_client.h]:
```suggestion
  /// Update resources of a node in GCS Service.
```

File: src/ray/rpc/gcs_server/gcs_rpc_client.h [hunk 3]
```diff
@@ -84,6 +84,18 @@ class GcsRpcClient {
...
+  VOID_RPC_CLIENT_METHOD(NodeInfoGcsService, UpdateResources, request, callback,
+                         node_info_grpc_client_)
+
+  /// Delete dynamic resources of node in GCS Service.
```
Comment (raulchen) [inline src/ray/rpc/gcs_server/gcs_rpc_client.h]:
```suggestion
  /// Delete resources of a node in GCS Service.
```

File: src/ray/gcs/gcs_server/node_info_handler_impl.cc [hunk 4]
```diff
@@ -110,5 +110,89 @@ void DefaultNodeInfoHandler::HandleReportBatchHeartbeat(
...
+
+void DefaultNodeInfoHandler::HandleDeleteResources(
+    const DeleteResourcesRequest &request, DeleteResourcesReply *reply,
+    SendReplyCallback send_reply_callback) {
+  ClientID node_id = ClientID::FromBinary(request.node_id());
+  std::vector<std::string> resource_names;
+  resource_names.reserve(request.resource_name_list_size());
+  for (int index = 0; index < request.resource_name_list_size(); ++index) {
+    resource_names.push_back(request.resource_name_list(index));
+  }
```
Comment (raulchen) [inline src/ray/gcs/gcs_server/node_info_handler_impl.cc]: there is a util function `MapFromProtobuf` for this.

**Review by raulchen — APPROVED** [review]
LGTM

File: src/ray/gcs/gcs_server/node_info_handler_impl.cc (hunk 4 above)
Comment (ffbin) [reply src/ray/gcs/gcs_server/node_info_handler_impl.cc]: fixed

File: src/ray/rpc/gcs_server/gcs_rpc_client.h (hunk 3 above)
Comment (ffbin) [reply src/ray/rpc/gcs_server/gcs_rpc_client.h]: fixed

File: src/ray/rpc/gcs_server/gcs_rpc_client.h (hunk 2 above)
Comment (ffbin) [reply src/ray/rpc/gcs_server/gcs_rpc_client.h]: fixed

File: src/ray/rpc/gcs_server/gcs_rpc_client.h (hunk 1 above)
Comment (ffbin) [reply src/ray/rpc/gcs_server/gcs_rpc_client.h]: fixed

**Review by raulchen — APPROVED** [review]

**Pull request merged**